test/unit/rules/no-unnecessary-curly-parens-test.js

```diff
@@ -0,0 +1,83 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-unnecessary-curly-parens',
+
+  config: true,
+
+  good: ['{{foo}}', '{{concat "a" "b"}}', '{{concat (capitalize "foo") "-bar"}}'],
```

Comment:
We should add and allow cases like `<Foo @bar={{(bar-helper)}} />`.
In the context of an argument or attribute, `{{(bar-helper)}}` and `{{bar-helper}}` are not the same thing.

Reply:
I can open another PR. Seems like this will reduce the cases we can confirm are unnecessary since we can't tell if `{{(this.something)}}` is a helper or not. That said most of the cases I see this actually happening in our codebase, the helper is being passed arguments so those can still be deemed unnecessary, right?

Reply:
Yup, that's right I think!

Reply:
https://github.com/ember-template-lint/ember-template-lint/pull/2800